### ISSUE TITLE
Apple Silicon support for `shellcheck`

### ISSUE THREAD

**Issue by vic1707** [jdx/rtx#892]
[Disclaimer]: Issue opened mainly for documentation for other users facing the same problem.
TL/DR: resolution will probably involve the merge of https://github.com/koalaman/shellcheck/pull/2736

---

**Describe the bug**
Just tried to install shellcheck with rtx, the logs were so fast i couldn't see there was a problem.
When trying to run shellcheck i get 
```
zsh ❯ shellcheck                                                
zsh: bad CPU type in executable: shellcheck
```

**To Reproduce**
Install shellcheck on a MacOS ARM based system

**Expected behavior**
Should install a ARM build.

**`rtx doctor` output**

```
rtx version:
  2023.9.0 macos-arm64 (2023-09-11)

build:
  Target: aarch64-apple-darwin
  Features: DEFAULT, NATIVE_TLS
  Built: Mon, 11 Sep 2023 12:16:04 +0000
  Rust Version: rustc 1.72.0 (5680fa18f 2023-08-23)
  Profile: release

shell:
  /bin/zsh
  zsh 5.9 (x86_64-apple-darwin22.0)

rtx data directory:
  /Users/vic1707/.local/share/rtx

rtx environment variables:
  RTX_SHELL=zsh

settings:
  {"always_keep_download": "false", "always_keep_install": "false", "asdf_compat": "false", "disable_default_shorthands": "false", "disable_tools": "[]", "experimental": "false", "jobs": "4", "legacy_version_file": "true", "legacy_version_file_disable_tools": "[]", "log_level": "INFO", "missing_runtime_behavior": "warn", "plugin_autoupdate_last_check_duration": "10080", "raw": "false", "trusted_config_paths": "[]", "verbose": "false", "yes": "false"}

config files:
  /Users/vic1707/.config/rtx/config.toml

plugins:
  bun          https://github.com/cometkim/asdf-bun.git#51d18b3
  go           (core)
  java         (core)
  jq           https://github.com/lsanwick/asdf-jq.git#f02032e
  node         (core)
  python       (core)
  ruby         (core)
  shellcheck   https://github.com/luizm/asdf-shellcheck.git#b160a52

toolset:
  bun@latest, go@latest, jq@latest, node@latest, python@latest, shellcheck@latest

No problems found
```

**Additional context**
I know shellcheck doesn't provide a darwin_aarch binary, but brew is able to install it, i wonder if binaries can be installed from there?
There is an open PR for upstream release of the aarch binary but its a WIP at the time of writing.
https://github.com/koalaman/shellcheck/pull/2736

**Comment by jdx** [comment]
belongs on https://github.com/luizm/asdf-shellcheck
